src/corehost/cli/deps_format.cpp

```diff
@@ -68,7 +63,7 @@ void deps_json_t::reconcile_libraries_with_targets(
                 size_t pos = library.first.find(_X("/"));
                 entry.library_name = library.first.substr(0, pos);
                 entry.library_version = library.first.substr(pos + 1);
-                entry.library_type = _X("package");
+                entry.library_type = pal::to_lower(library.second.at(_X("type")).as_string());
```

Comment:
Yes, the "by design" piece is what is happening today and with the fix. Except, in the bug case, the host ignores "project" dependencies in the app.deps.json when adding to TPA. It never broke anyone because of the APP_PATHS setting as the runtime found these in the app base.

I don't agree that there is no difference -- The type "project" tells that they are supposed to be found in the app base, not in the "packages" caches or FX for example, like I said this is an optimization and may even be correctness for the degenerate case of someone putting a project package in the cache.